starknet-providers/src/jsonrpc/models/mod.rs

```diff
@@ -396,47 +388,64 @@ pub struct TransactionReceiptMeta {
     #[serde_as(as = "UfeHex")]
     pub actual_fee: FieldElement,
     pub status: TransactionStatus,
-    /// Extra information pertaining to the status
-    pub status_data: Option<String>,
     #[serde_as(as = "UfeHex")]
     pub block_hash: FieldElement,
     pub block_number: u64,
-}
-
-/// Properties specific to invoke transaction (`INVOKE_TXN_RECEIPT_PROPERTIES`)
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct InvokeTransactionReceiptData {
+    #[serde(rename = "type")]
+    pub transaction_type: TransactionType,
     pub messages_sent: Vec<MsgToL1>,
-    /// In case this transaction was an L1 handler, this is the original message that invoked it
-    pub l1_origin_message: Option<MsgToL2>,
-    /// The events emitted as part of this transaction
     pub events: Vec<Event>,
 }
 
-/// Invoke transaction receipt
+/// Invoke Transaction Receipt (`INVOKE_TXN_RECEIPT`)
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct InvokeTransactionReceipt {
     #[serde(flatten)]
     pub meta: TransactionReceiptMeta,
+}
+
+/// Declare Transaction Receipt (`DECLARE_TXN_RECEIPT`)
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct DeclareTransactionReceipt {
+    #[serde(flatten)]
+    pub meta: TransactionReceiptMeta,
+}
+
+/// Deploy Account Transaction Receipt (`DEPLOY_ACCOUNT_TXN_RECEIPT`)
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct DeployAccountTransactionReceipt {
     #[serde(flatten)]
-    pub data: InvokeTransactionReceiptData,
+    pub meta: DeployTransactionReceipt,
+}
+
+/// Deploy Transaction Receipt (`DEPLOY_TXN_RECEIPT`)
+#[serde_as]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct DeployTransactionReceipt {
+    #[serde(flatten)]
+    pub meta: TransactionReceiptMeta,
+    /// The address of the deployed contract
+    #[serde_as(as = "UfeHex")]
+    pub contract_address: FieldElement,
 }
 
-/// A special type that covers both `DECLARE_TXN_RECEIPT` and `DEPLOY_TXN_RECEIPT` in the spec.
-/// This type exists because there's no way to distinguish between the 2 underlying types over the
-/// wire. The issue will be fixed in spec v0.2.0, upon which this type shall be removed.
+/// receipt for l1 handler transaction (`L1_HANDLER_TXN_RECEIPT`)
 #[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct DeclareOrDeployTransactionReceipt {
+pub struct L1HandlerTransactionReceipt {
     #[serde(flatten)]
     pub meta: TransactionReceiptMeta,
 }
 
-/// The `TXN_RECEIPT` type in the specification, except without the `PENDING_TXN_RECEIPT` variant.
+/// (`TXN_RECEIPT`)
 #[derive(Debug, Clone, Serialize, Deserialize)]
 #[serde(untagged)]
```

Comment:
This `untagged` used to be necessary, but it looks like the new spec version does tag the receipts, making the use of `untagged` here unnecessary. Change it to tagged.

Reply:
`untagged`s on all receipt types should be reviewed as they should no longer be necessary.